src/pubkey.ts

```diff
@@ -0,0 +1,209 @@
+const secp256k1 = require('secp256k1')
+const assert = require('assert')
+import BN = require('bn.js')
+import { toBuffer, setLength, setLengthLeft, bufferToHex } from './common'
+import { keccak } from './hash'
+
+export interface ECDSASignature {
+  v: number
+  r: Buffer
+  s: Buffer
+}
+
+/**
+ * Checks if the private key satisfies the rules of the curve secp256k1.
+ */
+export const isValidPrivate = function(privateKey: Buffer): boolean {
+  return secp256k1.privateKeyVerify(privateKey)
+}
```

Comment:
One side note: on refactoring we should go through all the file names, I think there are better versions for many of them (like this one, but also ``fromSigned``, ``baToJSON``, ``zeros``) being more descriptive and generally better grasp what the function is doing. Think this would help a lot on code readability on the usage side where people (we 😄) are integrating this within their code.